src/main/java/com/google/cloud/genomics/dataflow/pipelines/LoadReadsToBigQuery.java

```diff
@@ -0,0 +1,372 @@
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.util.gcsfs.GcsPath;
+import org.apache.beam.sdk.values.PCollection;
+import com.google.cloud.genomics.dataflow.readers.bam.BAMShard;
+import com.google.cloud.genomics.dataflow.readers.bam.ReadBAMTransform;
+import com.google.cloud.genomics.dataflow.readers.bam.Reader;
+import com.google.cloud.genomics.dataflow.readers.bam.ReaderOptions;
+import com.google.cloud.genomics.dataflow.readers.bam.ShardingPolicy;
+import com.google.cloud.genomics.dataflow.utils.GCSOptions;
+import com.google.cloud.genomics.dataflow.utils.GCSTempPathOptions;
+import com.google.cloud.genomics.dataflow.utils.GenomicsOptions;
+import com.google.cloud.genomics.dataflow.utils.ShardOptions;
+import com.google.cloud.genomics.utils.Contig;
+import com.google.cloud.genomics.utils.OfflineAuth;
+import com.google.cloud.genomics.utils.ShardBoundary;
+import com.google.common.base.Strings;
+import com.google.genomics.v1.Read;
+import com.google.genomics.v1.CigarUnit;
+
+import htsjdk.samtools.ValidationStringency;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * Pipeline for loading BAM file (hosted on GCS) into BigQuery.
+ *
+ */
+public class LoadReadsToBigQuery {
+
+  public static interface Options extends GCSOptions, ShardOptions, GCSTempPathOptions {
+
+    @Description("The Google Cloud Storage path to the BAM file to get reads data from.")
+    @Default.String("")
+    String getBAMFilePath();
+    void setBAMFilePath(String filePath);
+
+    @Description("For shareded BAM file input, the maximum size in megabytes of a shard processed by "
+        + "a single worker. A non-positive value implies no sharding.")
+    @Default.Integer(100)
+    int getMaxShardSizeMB();
+    void setMaxShardSizeMB(int maxShardSizeMB);
+
+    @Description("Whether to include unmapped mate pairs of mapped reads to match expectations "
+        + "of Picard tools.")
+    @Default.Boolean(false)
+    boolean isIncludeUnmapped();
+    void setIncludeUnmapped(boolean newValue);
+
+    @Description("BigQuery table (in [PROJECT]:[DATASET].[TABLE] format) to store the result.")
+    @Default.String("")
+    String getBigQueryTablePath();
+    void setBigQueryTablePath(String tablePath);
+
+    @Description("Whether to wait until the pipeline completes. This is useful for test purposes.")
+    @Default.Boolean(false)
+    boolean getWait();
+    void setWait(boolean wait);
+
+    public static class Methods {
```

Comment:
I don't really see the value in having a class to manage a single string parameter... It just means the reader has to chase this down.  Am I missing something?